ui/raidboss/data/triggers/dun_scaith.js

```diff
@@ -0,0 +1,483 @@
+            en: 'Buster on ' + + data.ShortName(matches[1]),
+          };
+        }
+      },
+    },
+    {
+      // Wailing Atomos is blue, Cursed Atomos is yellow.
+      // 1C9F:Aether is the circle AoE, 1CA0:Aetherial Chakram is the donut AoE
+      id: 'Dun Scaith Blue Atomos',
+      regex: / 14:(\y{AbilityCode}):\y{Name} starts using Juggling Sphere on Wailing Atomos/,
+      alertText: function(matches) {
+        if (matches[1] == '1C9F') {
+          return {
+            en: 'Avoid Untethered Blue',
+          };
+        }
+        if (matches[1] == '1CA0') {
+          return {
+            en: 'Go to Untethered Blue',
+          };
+        }
+      },
+    },
+    {
+      id: 'Dun Scaith Yellow Atomos',
+      regex: / 14:(\y{AbilityCode}):\y{Name} starts using Juggling Sphere on Cursing Atomos/,
+      alertText: function(matches) {
+        if (matches[1] == '1C9F') {
+          return {
+            en: 'Avoid Untethered Yellow',
+          };
+        }
+        if (matches[1] == '1CA0') {
+          return {
+            en: 'Go to Untethered Yellow',
+          };
+        }
+      },
+    },
+    {
+      id: 'Dun Scaith Blackfire',
+      regex: / 14:1CAA:Ferdiad Hollow starts using Blackfire/,
+      alertText: {
+        en: 'Avoid puddles',
+      },
+    },
+    {
+      id: 'Dun Scaith Debilitator',
+      regex: / 1A:........:\y{Name} gains the effect of (Fire|Water) Resistance Down/,
+      suppressSeconds: 10,
+      alertText: function(matches) {
+        if (matches[1] == 'Water') {
+          return {
+            en: 'Change puddles to fire',
+          };
+        }
+        if (matches[1] == 'Fire') {
+          return {
+            en: 'Change puddles to water',
+          };
+        }
+      },
+    },
+
+    // PROTO-ULTIMA
+
+    {
+      // The trident laser is a series of three separate casts
+      // Each has an incremental ID: 1D96, 1D97, 1D98
+      id: 'Dun Scaith Aetherochemical Laser',
+      regex: / 14:1D96:Proto Ultima starts using Aetherochemical Laser/,
+      alertText: {
+        en: 'Dodge trident laser',
+      },
+    },
+    {
+      // Handles both 1E52 Aetherochemical Flare and 1D9D Supernova
+      id: 'Dun Scaith Proto-Ultima Raid Damage',
+      regex: / 14:(?:1E52|1D9D): Proto Ultima Starts Using/,
+      condition: function(data) {
+        return data.role == 'healer';
+      },
+      alertText: {
+        en: 'Raid Damage',
+      },
+    },
+    {
+      id: 'Dun Scaith Prey Markers',
+      regex: / 1A:........:(\y{Name}) gains the effect of Prey/,
+      alertText: function(data, matches) {
+        if (data.me == matches[1]) {
+          return {
+            en: 'Prey--Avoid party and keep moving',
+          };
+        }
+      },
+    },
+    {
+      id: 'Dun Scaith Flare Star',
+      regex: / 14:1DA4:Proto Ultima starts using Flare Star/,
+      suppressSeconds: 1,
+      preRun: function(data) {
+        data.flareStarCount = (data.flareStarCount || 0) + 1;
```

Comment:
Yes. In a previous version of this trigger, I didn't think things through and attempted to set it up with both pre- and post-run data manipulations rather than just starting the condition index at 1 rather than 0.

As for the trigger text, I know it's a lot of words, but I'm not sure how to convey the correct mechanic resolution without at least most of them. I suppose it could be cut down to "wait for outside, then go" or something similar, but that doesn't indicate the proximity damage.